### ISSUE TITLE
Word layout ribbon (margins, orientation, size & columns) not identifying current option

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#5957]
In Word, on the Layout Ribbon, the first few options - Margins, Orientation, Size and Columns all have a list of options with an image representing each option and the text beside it. The current option for each (eg A4 paper) is visually highlighted but NVDA doesn't indicate anything. Also for margins, it reads the setting name (eg narrow) then the margin sizes for top, bottom, left and right, then reads the left and right values a second time.  This behaves consistently as you move through the options.

Tested in Word 2016 with NVDA 2016.1

**Comment by michaelDCurran** [comment]
NVDA  uses MSAA to access these list items. In this MSAA implementation, the selected item exposes the selected state, and the non-selected items do not... except for the focused item, which always exposes the selected state. This is definitely a bug in MS Office. However, its UIA implementation exposes the states correctly. For various reasons we decide not to use UIA in Office ribbons. We may need to review all this again, at least for later versions of Office. I can confirm that this issue also exists in Office 2013. It would be great if someone could provide developer info for these list items in Office 2010 if they exist.

**Comment by jcsteh** [comment]
Unfortunately, Word 2010 also exposes the selected state for the focused item, even if it's not actually selected.

**Comment by Adriani90** [comment]
This is still the case, even with UIA enabled.

**Comment by Adriani90** [comment]
In Word 2016 November 2022 Update and NVDA 2022.4, NVDA does not repeat the options anymore. In fact, also the not selected state is correctly announced, and the selected option is reported but without the selected state which is ok in my opinion as long as all other options are reported as not selected.

Only in the sections for line numbers and for hyphenation the states "enabled" or "disabled" are reported instead of "selected" or "not selected".

**Comment by Qchristensen** [comment]
Just testing with:
NVDA 2023.1.0.27913 
Windows 11 (64-bit) Version: 22H2, Build: 22621.1555 
Office 365 (64-bit) Version: 16.0.16227.20244 

I found the same as @Adriani90 - that is, moving through options on the layout ribbon (Margins, orientation, size, etc), the value and information is read out, and for everything EXCEPT the selected option, it reports "unselected".  EG with Margins, it reads "Normal Margins" and then goes on to read the size of each margin.  When you down arrow it says "Narrow margins not selected" and then reports what size the narrow margins are.

Reporting "selected" would make this slightly more obvious, but it is still possible to tell which option is selected once you know that everything else is reported as "not selected".

As noted, Line Breaks and Hyphenation report as "checked" or "not checked".  These options appear to be more traditional text options with a visual "tick" next to the selected item.  The other options mentioned (paper size etc) all have images depicting the options (rectangles either taller or wider for the orientation), a rectangle with blue lines indicating how wide the margins are etc)

**Comment by Adriani90** [comment]
Given that selected items in Windows generally are not reported with the word "selected" (e.g. file explorer), this experience is common and is not wrong. Only elements that are not selected are reported with the word "not selected".
I am closing this one as won't fix. If you are still having arguments why the current behavior should change, please comment and we can reopen.

**Comment by CyrilleB79** [comment]
> Given that selected items in Windows generally are not reported with the word "selected" (e.g. file explorer), this experience is common and is not wrong.

Actually there is an important difference between the experience in Word ribbons here and in Windows Explorer. In Windows Explorer, as in many other lists, moving with arrows moves the selection. Actually, we use arrows to move the selection. That's the reason why we do not bother reporting "selected" in lists because moving with arrows is the most common move and it guarantees that the focused item is selected. In Word ribbons instead, moving with arrows does not change the selection; Enter on an item does. Actually, here, we use arrows to know which element is selected, not only to select a new one.

> I am closing this one as won't fix. If you are still having arguments why the current behavior should change, please comment and we can reopen.

In addition to what I have written above, @Qchristensen already provided an argument:
> Reporting "selected" would make this slightly more obvious, but it is still possible to tell which option is selected once you know that everything else is reported as "not selected".


He also adds:

> As noted, Line Breaks and Hyphenation report as "checked" or "not checked". These options appear to be more traditional text options with a visual "tick" next to the selected item. The other options mentioned (paper size etc) all have images depicting the options (rectangles either taller or wider for the orientation), a rectangle with blue lines indicating how wide the margins are etc)

Actually, in the first place, that's the way these ribbons have been designed which is quite bad, i.e. we should have checked / unchecked instead of selected / unselected here. Anyway, NVDA has to cop with it.

To summarize, in this case, NVDA should explicitly report "selected".

Re-opening.
